pygmt/tests/test_figure.py

```diff
@@ -92,6 +92,46 @@ def test_figure_savefig_exists():
         fname.unlink()
 
 
+def test_figure_savefig_geotiff():
+    """
+    Make sure .tif generates a normal TIFF file and .tiff generates a GeoTIFF
+    file.
+    """
+    fig = Figure()
+    fig.basemap(region=[0, 10, 0, 10], projection="M10c", frame=True)
+
+    # Save as GeoTIFF
+    geofname = Path("test_figure_savefig_geotiff.tiff")
+    fig.savefig(geofname)
+    assert geofname.exists()
+    assert geofname.with_suffix(".pgw").exists()  # The companion world file
```

Comment:
Strange, why is a pgw file generated instead of a tfw file? See https://en.wikipedia.org/wiki/World_file#Filename_extension. A .tif or .tiff file should be linked to .tfw no?

Reply:
See https://github.com/GenericMappingTools/pygmt/issues/2658#issuecomment-1712392325 for some tests.

Here, we call `gmt psconvert -A -P -W+g` without specifying the `-T` option. 

Looking at the `psconvert` source code (https://github.com/GenericMappingTools/gmt/blob/master/src/psconvert.c#L1008-L1010), 
```
	if (!Ctrl->T.active) {	/* Set default output device if none is specified */
		Ctrl->T.device = (Ctrl->W.warp) ? GS_DEV_PNG : GS_DEV_JPG;	/* Lossless PNG if we are making a geotiff in the end */
	}
```
The default format is PNG is `-W+g` is used. I think GMT fist converts PS to lossless PNG, then write the world file, and calls `gdal_translate` to combine the PNG and world file into a GeoTIFF file. The extension of the world file is determined from the extension of the raster file (in this case PNG), thus we have a `pgw` file. This may be an upstream feature or bug.

Actually, since we already have a `GeoTIFF` file, the world_file is no longer needed. Maybe we should remove it?

Reply:
Ah ok, thanks for the explanation. You're right that we wouldn't need the `pgw` file or any world file anyway since the GeoTIFF should already have the metadata embedded inside.

> Maybe we should remove it?

Maybe best to add the code to remove the `pgw` file upstream in GMT when a GeoTIFF is created, or at least rename `pgw` to `tfw`? On the PyGMT side, we could match the upstream implementation with GMT <= 6.4.0.

Reply:
> Maybe best to add the code to remove the `pgw` file upstream in GMT when a GeoTIFF is created, or at least rename `pgw` to `tfw`?

Removing `pgw` makes more sense to me. I'll open an issue report and see what the GMT team thinks about it.

> On the PyGMT side, we could match the upstream implementation with GMT <= 6.4.0.

Do you mean we should keep the `pgw` file? I'm thinking about adding a new parameter `worldfile=True` to `Figure.savefig()` which generates a companion world file.

Reply:
> > On the PyGMT side, we could match the upstream implementation with GMT <= 6.4.0.
> 
> Do you mean we should keep the `pgw` file? I'm thinking about adding a new parameter `worldfile=True` to `Figure.savefig()` which generates a companion world file.

I mean we should match whatever GMT does with the `pgw` world file later in GMT 6.5.0, but open up that issue first. The `worldfile=True` parameter sounds nice actually, this would be similar to GDAL's `TFW=YES/NO` option when creating GeoTIFFs - https://gdal.org/drivers/raster/gtiff.html#creation-options

Reply:
Note that https://github.com/GenericMappingTools/gmt/pull/7865 was merged, and the upcoming GMT 6.5.0 should delete the `pgw` world file if a GeoTIFF output is requested.